src/material/datepicker/datepicker-input.ts

```diff
@@ -112,8 +112,12 @@ export class MatDatepickerInput<D> extends MatDatepickerInputBase<D | null, D>
   @Input('matDatepickerFilter')
   get dateFilter() { return this._dateFilter; }
   set dateFilter(value: DateFilterFn<D | null>) {
+    const wasMatchingFilter = this._matchesFilter(this.value);
```

Comment:
To keep the naming in line with the range picker (`wasMatchingStart`, `wasMatchingEnd`) I would call this `wasMatchingValue`